router/src/main/java/io/micronaut/web/router/exceptions/UnsatisfiedRouteExceptionFactory.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2017-2019 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.web.router.exceptions;
+
+import io.micronaut.core.bind.annotation.Bindable;
+import io.micronaut.core.type.Argument;
+import io.micronaut.http.annotation.*;
+
+import java.lang.annotation.Annotation;
+import java.util.Optional;
+
+/**
+ * A factory for creating specialized UnsatisfiedRouteException.
+ *
+ * @author Fabien Renaud
+ * @since 1.3.0
+ */
+public final class UnsatisfiedRouteExceptionFactory {
+
+    private UnsatisfiedRouteExceptionFactory() {
+    }
+
+    /**
+     * Creates a specialized UnsatisfiedRouteException given the provided argument.
+     *
+     * @param argument The {@link Argument}
+     * @return A UnsatisfiedRouteException
+     */
+    public static UnsatisfiedRouteException create(Argument<?> argument) {
+        Optional<Class<? extends Annotation>> classOptional = argument.getAnnotationMetadata().getAnnotationTypeByStereotype(Bindable.class);
+
+        if (classOptional.isPresent()) {
+            Class<? extends Annotation> clazz = classOptional.get();
+            Optional<Object> valOptional = argument.getAnnotationMetadata().getValue(clazz);
+            String name = valOptional.orElse(argument).toString();
```

Comment:
Please use `String name = argument.getAnnotationMetadata().stringValue().orElse(null)`

Reply:
done

Reply:
Undid that.

Using `argument.getAnnotationMetadata().stringValue(clazz).orElse(null)` returned a null name for a bunch of tests (and broke them) so it's not equivalent to current implementation...

https://travis-ci.org/micronaut-projects/micronaut-core/builds/614591730?utm_source=github_status&utm_medium=notification